ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,192 @@
+    bool exists = false;
+
+    std::string obj_prefix = randomize_test_prefix();
+
+    azure_connection conn = azure_connection("myblobcontainer1", obj_prefix);
+    azure_connection conn_bad = azure_connection("myblobcontainer1", "bad_prefix_");
+
+    std::vector<std::pair<std::string, std::string>> blob_objects;
+
+    const std::string object_name = "test_object";
+    const std::string object_name_2 = "test_object_2";
+    const std::string non_exist_object_key = "test_non_exist";
+
+    // Payloads for blobs uploaded to container.
+    const std::string payload = "payload";
+    const std::string payload_2 = "Testing offset and substring";
+    blob_objects.push_back(std::make_pair(object_name, payload));
+    blob_objects.push_back(std::make_pair(object_name_2, payload_2));
+
+    // Add objects into the container.
+    for (auto pair : blob_objects) {
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + pair.first);
+        blob_client.UploadFrom(create_file(pair.first, pair.second));
+    }
+
+    SECTION("Check Azure connection constructor.", "[azure-connection]")
+    {
+        REQUIRE_THROWS_WITH(azure_connection("Bad_bucket", ""), "Bad_bucket : No such bucket.");
+    }
+
+    SECTION("Check object exists in Azure.", "[azure-connection]")
+    {
+        // Object exists so there should be 1 object.
+        REQUIRE(conn.object_exists(object_name, exists) == 0);
+        REQUIRE(exists == true);
+
+        // Object does not exist so there should be 0 objects.
+        REQUIRE(conn.object_exists(non_exist_object_key, exists) == 0);
+        REQUIRE(exists == false);
+    }
+
+    SECTION("List Azure objects under the test bucket.", "[azure-connection]")
+    {
+        std::vector<std::string> objects;
+
+        // No matching objects. Object size should be 0.
+        REQUIRE(conn.list_objects(obj_prefix + non_exist_object_key, objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // No matching objects with list single functionality. Object size should be 0.
+        REQUIRE(conn.list_objects(obj_prefix + non_exist_object_key, objects, true) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // List all objects. Object size should be 2.
+        REQUIRE(conn.list_objects(obj_prefix + object_name, objects, false) == 0);
+        REQUIRE(objects.size() == blob_objects.size());
+        objects.clear();
+
+        // List single. Object size should be 1.
+        REQUIRE(conn.list_objects(obj_prefix + object_name, objects, true) == 0);
+        REQUIRE(objects.size() == 1);
+
+        objects.clear();
+    }
+
+    SECTION("Test delete functionality for Azure.", "[azure-connection]")
+    {
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + object_name + "1");
+
+        blob_client.UploadFrom(create_file(object_name + "1", payload));
+
+        // Test that an object can be deleted.
+        REQUIRE(conn.delete_object(object_name + "1") == 0);
```

Comment:
We should check that the delete object actually deletes the object from the cloud.

Reply:
done